nuxeo-runtime/nuxeo-runtime-jtajca/src/main/java/org/nuxeo/runtime/transaction/TransactionHelper.java

```diff
@@ -602,4 +616,55 @@ public static void runInTransaction(Runnable runnable) {
         }
     }
 
+    /**
+     * Calls the given {@link Supplier} in a context without a transaction. The supplier must do its own transactional
+     * cleanup to restore the thread to a pristine state.
+     *
+     * @param supplier the {@link Supplier}
+     * @return the supplier's result
+     * @since 11.1
+     */
+    protected static <R> R runWithoutTransactionInternal(Supplier<R> supplier) {
+        // if there is already no transaction, run in this thread
+        if (isNoTransaction()) {
+            return supplier.get();
+        }
+        // otherwise use a separate thread to get a separate transactional context
+        try {
+            return EXECUTOR.submit(() -> supplier.get()).get();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt(); // restore interrupted status
+            throw new RuntimeException(e);
+        } catch (ExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof RuntimeException) {
+                throw (RuntimeException) cause;
+            } else {
+                throw new RuntimeException(cause);
+            }
+        }
+    }
+
+    /**
+     * Calls the given {@link Supplier} and cleans up the transaction context afterwards.
+     *
+     * @param supplier the {@link Supplier}
+     * @return the supplier's result
+     * @since 11.1
+     */
+    protected static <R> R runAndCleanupTransactionContext(Supplier<R> supplier) {
+        try {
+            return supplier.get();
+        } finally {
+            if (!TransactionHelper.isNoTransaction()) {
+                // restore the no-transaction context of this thread
+                try {
+                    TransactionHelper.commitOrRollbackTransaction();
```

Comment:
You can call directly the method without `TransactionHelper` right?

Reply:
Ah yes indeed, force of habit... I'll change that